src/main/java/net/neoforged/neoforge/transfer/resources/ItemResource.java

```diff
@@ -0,0 +1,337 @@
+        return without(type.get());
+    }
+
+    @Override
+    public DataComponentMap getComponents() {
+        return innerStack.immutableComponents();
+    }
+
+    @Override
+    public DataComponentPatch getComponentsPatch() {
+        return innerStack.getComponentsPatch();
+    }
+
+    /**
+     * Creates an {@link ItemStack} of the specified count.
+     *
+     * @param count The amount of the item the stack should have. Must be non-negative.
+     * @return A new copy of the inner item stack with the specified count.
+     * @throws IllegalArgumentException when count is negative.
+     */
+    public ItemStack toStack(int count) {
+        TransferPreconditions.checkNonNegative(count);
+        if (count == 0) return ItemStack.EMPTY;
+        return this.innerStack.copyWithCount(count);
+    }
+
+    /**
+     * Creates an {@link ItemStack} with a count of 1.
+     *
+     * @return A new copy of the inner item stack with a count of 1.
+     */
+    public ItemStack toStack() {
+        return this.innerStack.copyWithCount(1);
+    }
+
+    /**
+     * Creates a list of {@link ItemStack ItemStacks} of the specified count taking into account the max stack size of the item.
+     * Note: This is not guaranteed to be mutable. For the case of empty, an immutable empty list is returned.
+     *
+     * @param count The amount of the item the stack should have. Must be non-negative.
+     * @return A list of item stacks that all have a maximum count of the max stack size of the item.
+     * @throws IllegalArgumentException when the count is negative.
+     */
+    public List<ItemStack> toStacks(int count) {
+        TransferPreconditions.checkNonNegative(count);
+        if (count == 0 || isEmpty())
+            return Collections.emptyList();
+
+        int maxStackSize = getMaxStackSize();
+        int stackCount = count / maxStackSize;
+        int remainder = count % maxStackSize;
+
+        List<ItemStack> stacks = new ArrayList<>(stackCount + 1);
+        var stack = toStack(maxStackSize);
+        for (int i = 0; i < stackCount; i++) {
+            stacks.add(stack.copy());
+        }
+
+        if (remainder > 0) {
+            stacks.add(stack.copyWithCount(remainder));
+        }
+        return stacks;
+    }
+
+    public int getMaxStackSize() {
+        return innerStack.getMaxStackSize();
+    }
+
+    /**
+     * Returns the hover name of the {@link ItemStack}.
+     *
+     * @return The hover name of the {@link ItemStack}
+     */
+    public Component getHoverName() {
+        return innerStack.getHoverName();
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) return true;
+        return obj instanceof ItemResource other && other.is(innerStack);
+    }
+
+    @Override
+    public int hashCode() {
+        return ItemStack.hashItemAndComponents(innerStack);
+    }
+
+    @Override
+    public String toString() {
+        return innerStack.getItem().toString();
```

Comment:
I think we can do a little better here, at least to indicate how many data components are patched.